src/main/java/life/qbic/model/download/Authentication.java

```diff
@@ -0,0 +1,62 @@
+package life.qbic.model.download;
+
+import ch.ethz.sis.openbis.generic.asapi.v3.IApplicationServerApi;
+import ch.systemsx.cisd.common.spring.HttpInvokerUtils;
+
+public class Authentication {
+
+    private static String sessionToken;
+    private String user;
+    private String password;
+    private final IApplicationServerApi applicationServer;
+
+
+    public Authentication(
+            String user,
+            String password,
+            String AppServerUri) {
+        this.user = user;
+        this.password = password;
+        if (!AppServerUri.isEmpty()) {
+            this.applicationServer =
+                    HttpInvokerUtils.createServiceStub(
+                            IApplicationServerApi.class, AppServerUri + IApplicationServerApi.SERVICE_URL, 10000);
+        } else {
+            this.applicationServer = null;
+        }
+        this.setCredentials(user, password);
+    }
+
+    /**
+     * Login method for openBIS authentication
+     *
+     * returns 0 if successful, 1 else
+     */
+    public void login() throws ConnectionException, AuthenticationException {
+        try {
+            sessionToken = applicationServer.login(user, password);
+        } catch (Exception e) {
```

Comment:
Do we need to logout somewhere? I could not find anything in the old code so I assume we don't have to :shrug:

Reply:
I can do some research about that:)

Reply:
Until now, we did not logout. But I found a logout method for the application server and added it at the end of the whole process... or is that unnecessary?